File: Task002.py
#Silalak Kaewjanpet 610510670

def PutThingsIntoToTheBag(things,W):
    weight = W
    count = len(things)
    price = 0
    for i in range(count):
      
        if (weight == 0 or i == count):
            break
        elif (weight >= things[i][1]):
            price += things[i][0]
            weight -= things[i][1]
        else:
            price += (things[i][0]/things[i][1])*weight
            weight -= weight

    return price


def main():
    things = []
    N, W = map(int,input().split())
    num = list(map(int, input().split()))
    things = list(zip(num[::2], num[1::2]))
    things.sort(key = lambda x: x[0] / x[1])
    things.reverse()
    print("%.2f"% PutThingsIntoToTheBag(things,W))
    
if __name__ == "__main__":
    main()

File: Task002.c
//Silalak Kaewjanpet sec 001
//610510670

#include <stdio.h>
//#include <conio.h>


void PutInBag() 
{
    int N, W, c = 0;            //N = จำนวนสิ่งของ W = น้ำหนักที่รับได้
    int weight; 
    float val = 0;
    scanf("%d,%d", &N, &W); 
    int object[N*2];
    int i;
    for(i=0; i<N*2; i++)
    {
        scanf("%d",&object[i]); 
    }

    for(i=0; i<N*2; i=i+2)
    {
        if(object[i+1] < W)
        {
            val += object[i];
            W = W - object[i+1];         
        }
        else
        {
            val += ((object[i]*W)/object[i+1]);
        }
        
    }
    printf("%.2f",&val);

    
}




int main()
{
    PutInBag();
    
}